### ISSUE TITLE
Swift 6.0 support

### ISSUE THREAD

**Issue by lhoward** [jeremy-prater/meta-swift#6]
Are there any plans to add support for Swift 5.9?

**Comment by jeremy-prater** [comment]
Does it work in linux correctly, and how does spm work now? `swift package manger` work now

**Comment by lhoward** [comment]
I’m probably a few months away from getting into Yocto but I shall file one if no-one else does. Swift 5.9 Linux is fine for 64-bit at least.

**Comment by jeremy-prater** [comment]
It's the swift on arm v7 32-bit that gets complicated.

**Comment by xtremekforever** [comment]
It may not be so bad that the compiler doesn't work on arm32 anymore. I find that cross-compiling is the best way to do things, and also starting with Swift 5.9 we have the ability to create our own cross-compilation SDKs. However, I wonder if we have the following artifacts for armv7:

https://github.com/xtremekforever/swift-armv7/releases

If they can be included in this meta-swift layer and used, or if something else is needed for it to work for Yocto?

**Comment by lhoward** [comment]
Ooh, thanks for that link – very handy (although I've kind of committed to 64-bit for now, and moved to Swift 6.0.2).

I will likely start looking at Yocto for our product in this coming January so I will have some cycles to contribute to this.

**Comment by xtremekforever** [comment]
Not to worry- I'm doing this for my Raspberry Pi and other armv7 boards that run Linux. I have basically realized that supporting armv7 officially in swiftlang/swift will either take a long time or maybe never happen, but supporting it via third-party scripts and builds is easy enough starting with Swift 5.9 when using @colemancda's scripts. 

Sounds good to me!

**Comment by lhoward** [comment]
Well switching to a Pi4 or Pi5 will free you from armv7 ;) (I'm also using Pi here.)

**Comment by colemancda** [comment]
@xtremekforever It will never happen, we've been supporting this since 2018 I think and Apple simply has no interest, every release breaks something. On the up side, it makes this project really fun haha. Android Armv7 is much more stable, they even have the compiler working. Cross compilation is definitely the way to go for embedded linux.

**Comment by xtremekforever** [comment]
Yeah I think that's true especially since Fedora, RHEL, and Amazon Linux don't support armv7. So there's no way they will accept official support for Linux armv7 since half of the supported distributions don't even support the architecture anymore. However, I still see that having some runtime support for armv7 distros and cross compiling is definitely the way 👌🏻

**Comment by lhoward** [comment]
Also, reasonably priced FPGA hard cores will be armv7 for a while yet I imagine.

**Comment by lhoward** [comment]
I'm working on updating for Swift 6.0.2.

One question: in `swift-stdlib.bb`, why are the CMake variables (e.g. `SWIFT_BUILD_SOURCEKIT`) defined thrice? Can we just do it in one place?

**Comment by lhoward** [comment]
New PR #11

**Comment by xtremekforever** [comment]
We finally have support for 6.0 and later!
